src/main/java/org/candlepin/subscriptions/inventory/db/model/InventoryHost.java

```diff
@@ -20,30 +20,53 @@
  */
 package org.candlepin.subscriptions.inventory.db.model;
 
-import com.vladmihalcea.hibernate.type.json.JsonBinaryType;
-
-import org.hibernate.annotations.Type;
-import org.hibernate.annotations.TypeDef;
-
 import java.io.Serializable;
 import java.time.OffsetDateTime;
-import java.util.Map;
 import java.util.UUID;
 
-import javax.persistence.Basic;
 import javax.persistence.Column;
+import javax.persistence.ColumnResult;
+import javax.persistence.ConstructorResult;
 import javax.persistence.Entity;
-import javax.persistence.FetchType;
 import javax.persistence.Id;
+import javax.persistence.NamedNativeQuery;
+import javax.persistence.SqlResultSetMapping;
 import javax.persistence.Table;
 
 
 /**
  * Represents a host entity stored in the inventory service's database.
  */
+@SuppressWarnings({"indentation", "linelength"})
 @Entity
 @Table(name = "hosts")
-@TypeDef(name = "jsonb", typeClass = JsonBinaryType.class)
+@SqlResultSetMapping(
+    name = "inventoryHostFactsMapping",
+    classes = {
+        @ConstructorResult(
+            targetClass = InventoryHostFacts.class,
+            columns = {
+                @ColumnResult(name = "account"),
+                @ColumnResult(name = "display_name"),
+                @ColumnResult(name = "org_id"),
+                @ColumnResult(name = "cores"),
+                @ColumnResult(name = "sockets"),
+                @ColumnResult(name = "is_rhel"),
+                @ColumnResult(name = "products"),
+                @ColumnResult(name = "sync_timestamp")
+            }
+        )
+    }
+)
+@NamedNativeQuery(name = "InventoryHost.getFacts",
+    query =
+        "select " +
+            "account, display_name, facts->'rhsm'->>'orgId' as org_id, facts->'rhsm'->>'CPU_CORES' as cores, " +
```

Comment:
I have never seen this arrow notation before.  Is this a Spring Data thing?  Do you have a link explaining it handy?

Reply:
Ah, I see.  JSON dereferencing.  Cool.